### ISSUE TITLE
Error: ClassNLLCriterion.lua:44: Assertion `cur_target >= 0 && cur_target < n_classes' failed.

### ISSUE THREAD

**Issue by sowjanya143** [torch/nn#471]
trainer:train(trainset)
It worked for a week without any errors. Overnight, I have no idea what happened I am unable to train the network. Please help..!

**Comment by sowjanya143** [comment]
confusion matrix just prints 0s all the time and the training progress strucks at 1/(traindata size).
I used this template for multiclass using googlenet. can someone help me?

**Comment by cjds** [comment]
@sowjanya143 what was the error? I'm getting the same one

**Comment by soumith** [comment]
@cjds your target class is out of range. If you have 10 classes as output of your network, your groundtruth class seems to be < 1 or > 10

**Comment by cjds** [comment]
Yep. Thanks @soumith. Figured it out

**Comment by holgafreak** [comment]
I'm trying to classify images, and as very new to lua and torch, took code from the demos (ltrain-face-detector and load-data), using the cifar cats and dogs images. After hours of hunting, the alley_cat_s_00085.png in the train dataset gave this error. When I removed the image, my code (hooray!) works. The labels when the cat image was present went from negative to 2e+27 (one test print showed this) Otherwise training and testing goes ok, even with very small subset of the train and test imageset. I have no idea, where things go haywire, but the cat image itself opens nicely on preview on OS X or showell on Linux, Both torch-cl on OS X and torch on Linux give the same error on this specific image.

**Comment by joelteply** [comment]
I am a C++ programmer and naturally was using 0 indexes for the labels by mistake. Maybe someone also made that mistake

**Comment by apaszke** [comment]
Actually you can compile all TH* libraries with -DTH_INDEX_BASE=0, and it will use zero for indexing operations

**Comment by greed2411** [comment]
this forum [post](https://discuss.pytorch.org/t/solved-assertion-cur-target-0-cur-target-n-classes-failed/8034/2?u=greed2411) helped me.

**Comment by Sedherthe** [comment]
> @cjds your target class is out of range. If you have 10 classes as output of your network, your groundtruth class seems to be < 1 or > 10

@cjds So is 1 indexing the right way to go or 0 indexing?
